lib/internal/util/types.js

```diff
@@ -54,8 +58,16 @@ function isBigUint64Array(value) {
   return TypedArrayPrototypeGetSymbolToStringTag(value) === 'BigUint64Array';
 }
 
+function isArrayBufferDetached(value) {
+  if (value instanceof ArrayBuffer && ArrayBufferPrototypeGetByteLength(value) === 0) {
```

Comment:
I wouldn't be surprised if that `instanceof` check would have a dramatic impact on performance.

Reply:
Referencing: https://github.com/RafaelGSS/nodejs-bench-operations/blob/main/RESULTS-v19.md#comparison-using-instanceof

I can change it to `value != null && ...` , and expect `.byteLength` to be the differentiator.

Reply:
I think the best course of action is to make an internal `isArrayBufferDetached` without any instanceof check, and if we have a public one, it would return `value instanceof ArrayBuffer && internal.isArrayBufferDetached(value)`.

Reply:
@aduh95 https://github.com/nodejs/node/pull/45568

Reply:
also, instanceof is brittle and can be forged, and doesn’t work cross-realm. In JS the way to check this is using the byteLength getter in a try/catch.